src/pages/MoviesPage.vue

```diff
@@ -1,10 +1,86 @@
 <template>
-  <MainTitle text="This is Movies Page" />
+<section>
+  <MainTitle text="All the Movies" />
+  <div class="searchBar">
+    <BaseInput  
+      type="text"
+      label="search"
+      placeholder="Search for..."
+      v-model="query"/>
+  </div>
+  <div class="movies-container">
+     
+     <MovieCard
+      v-for="movie in movies"
+      :key="movie.id"
+      :movie="movie"
+    />
+
+   
+  </div>
+
+  
+</section>
+  
 </template>
 
 <script>
 import MainTitle from "../components/MainTitle.vue";
+import MovieCard from '../components/MovieCard.vue';
+import BaseInput from '../components/BaseInput.vue';
+import axios from 'axios';
+
 export default {
-  components: { MainTitle },
+  components: { MainTitle, MovieCard, BaseInput },
+
+  data(){
+  return {
+    movies: [],
+    errorMsg: '',
+    query: '',
+  }
+  },
+
+  computed:{
+    findMovies(){
+    const foundMovies = this.movies.filter(movie => movie.title.toLowerCase()
+    .includes(this.query.toLowerCase()));
+     return foundMovies;
+    }
+  },
+
+  methods:{
+    async fetchData(){
+      try{
+         const response = await axios.get('http://localhost:3000/movies');
```

Comment:
You can set axios base url as global value so in future when it will change you'll need to change it only in one place.
https://axios-http.com/docs/config_defaults